sos/plugins/logs.py

```diff
@@ -50,83 +39,31 @@ def setup(self):
-        )
-        self.do_path_regex_sub(
-            r"/etc/rsyslog*",
-            r"pwd=.*",
-            r"pwd=[******]"
-        )
-
-
-class RedHatLogs(Logs, RedHatPlugin):
-
-    option_list = [
-        ("log_days", "the number of days logs to collect", "", 3)
-    ]
-
-    def setup(self):
-        super(RedHatLogs, self).setup()
-        # NOTE: for historical reasons the 'messages' and 'secure' log
-        # paths for the RedHatLogs plugin do not obey the normal --all-logs
-        # convention. The rotated copies are collected unconditionally
-        # due to their frequent importance in diagnosing problems.
-        messages = "/var/log/messages"
-        secure = "/var/log/secure"
-
-        have_messages = os.path.exists(messages)
-
-        self.add_copy_spec(secure + "*")
-        self.add_copy_spec(messages + "*")
-
-        # collect three days worth of logs by default if the system is
-        # configured to use the journal and not /var/log/messages
-        if not have_messages and self.is_installed("systemd"):
-            try:
-                days = int(self.get_option("log_days"))
-            except ValueError:
-                days = 3
-            if self.get_option("all_logs") and self.get_options("since"):
-                since = self.get_options("since")
-            else:
-                since = "-%ddays" % days
-            self.add_journal(since=since)
-
-
-class DebianLogs(Logs, DebianPlugin, UbuntuPlugin):
-
-    option_list = [
-        ("log_days", "the number of days logs to collect", "", 7)
-    ]
-
-    def setup(self):
-        super(DebianLogs, self).setup()
-        days = int(self.get_option("log_days"))
-        journal = os.path.exists("/var/log/journal/")
         self.add_copy_spec([
-            "/var/log/udev",
-            "/var/log/dist-upgrade",
+            "/etc/syslog.conf",
+            "/etc/rsyslog.conf",
+            "/etc/rsyslog.d",
+            "/var/log/boot.log",
             "/var/log/installer",
-            "/var/log/unattended-upgrades"
-            ])
+            "/var/log/unattended-upgrades",
+        ])
+
+        self.add_cmd_output("journalctl --disk-usage")
         self.add_cmd_output('ls -alRh /var/log/')
 
+        journal = os.path.exists("/var/log/journal/")
         if journal and self.is_installed("systemd"):
+            self.add_journal()
+            self.add_journal(boot="this", catalog=True)
+            self.add_journal(boot="last", catalog=True)
             if self.get_option("all_logs"):
-                since = self.get_options("since")
-            else:
-                since = "-%ddays" % days
-            self.add_journal(since=since)
-        else:
+                 self.add_copy_spec("/var/log/journal/*")
+        else:  # If not using journal
+            # Red Hat was collecting in all cases
+            self.add_copy_spec([
+                    "/var/log/messages*",
+                    "/var/log/secure*",
+            ])
```

Comment:
Is it OK if we collect `messages*` and `secure` *only* when not journal and systemd? (i.e. basically on legacy systems only), @TurboTurtle ?

(my personal preference is to collect these any time, but I am OK in dropping that as duplicate info to journal logs)

Reply:
IMO we should capture those every time.

Reply:
Will add these to general collection then.  Also expanding our syslog/kern/auth to do 1 more rotated log.

Reply:
*to non-conditional collection.